genie-agent/src/main/java/com/netflix/genie/agent/execution/exceptions/AgentRegistrationException.java

```diff
@@ -0,0 +1,46 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.exceptions;
+
+/**
+ * Error to register the agent with the server.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+public class AgentRegistrationException extends Exception {
```

Comment:
Are you SURE you don't want to just go with RuntimeExceptions? I say this from experience of driving myself nuts putting `@throws GenieException` on every server side method just in case we ever through an exception and most of the time just propagating it all the way up to ExceptionMapper anyway cause they're unrecoverable. In this case is it not also unrecoverable? There's nothing stopping you from putting `@throws` in your JavaDoc for these runtime exceptions I Just think you're asking to add a lot of boilerplate with no real benefit